compiled_starters/go/.codecrafters/compile.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+#
+# This script is used to compile your program on CodeCrafters
+# 
+# This runs before .codecrafters/run.sh
+#
+# Learn more: <insert-link>
+
+go build -o /tmp/shell-target cmd/myshell/*.go
```

Comment:
All `compile.sh` scripts need `set -e` at the beginning. Else a failure won't cause a non-zero exit code, and we won't know when the compile step fails.